test/plugin/scenarios/kafka-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/kafka/controller/CaseController.java

```diff
@@ -223,29 +235,33 @@ public void run() {
             consumerProperties.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
             KafkaConsumer<String, String> consumer = new KafkaConsumer<>(consumerProperties);
             consumer.subscribe(topicPattern, new NoOpConsumerRebalanceListener());
-            int i = 0;
-            while (i++ <= 10) {
-                try {
-                    Thread.sleep(1 * 1000);
-                } catch (InterruptedException e) {
-                }
+            while (true) {
+                if (pollAndInvoke(consumer)) break;
+            }
+            consumer.close();
+        }
 
-                ConsumerRecords<String, String> records = consumer.poll(100);
+        @KafkaPollAndInvoke
+        private boolean pollAndInvoke(KafkaConsumer<String, String> consumer) {
```

Comment:
Are you forcing to create 2 consumer methods?

Reply:
yes